cosmwasm/contracts/pyth/src/contract.rs

```diff
@@ -182,9 +201,18 @@ fn execute_governance_instruction(
     }
 
     let response = match instruction.action {
-        UpgradeContract { .. } => {
-            // FIXME: implement this
-            Err(PythContractError::InvalidGovernancePayload)?
+        UpgradeContract { address } => {
+            if instruction.target_chain_id == 0 {
+                Err(PythContractError::InvalidGovernancePayload)?
+            }
+
+            // Hack: We're using the first 8 bytes of the address field as the new code id.
```

Comment:
Sorry for the inconsistency here.. for this one aptos uses 32 bytes and Eth uses 20 bytes. I think it is fine to define the governance instruction to use only 8 bytes instead of getting a prefix of a longer address. 

Although not used anywhere, you will need to define it here too (for scripts/tools that manage deployments):
https://github.com/pyth-network/pyth-crosschain/blob/main/third_party/pyth/xc-governance-sdk-js/src/instructions.ts#L103-L121